code/Model/SiteTree.php

```diff
@@ -2747,7 +2754,8 @@ public function getTreeTitle()
         $children = $this->creatableChildren();
         $flags = $this->getStatusFlags();
         $treeTitle = sprintf(
-            "<span class=\"jstree-pageicon page-icon class-%s\"></span><span class=\"item\" data-allowedchildren=\"%s\">%s</span>",
+            "<span class=\"jstree-pageicon page-icon %s class-%s\"></span><span class=\"item\" data-allowedchildren=\"%s\">%s</span>",
+            $this->config()->get('icon') ? '' : $this->config()->get('icon_class'),
```

Comment:
I feel like this deserves to be in its own method, e.g. `getIconClass()` or something, but it's not a blocker